test/fixtures/server.js

```diff
@@ -26,33 +32,223 @@ export async function createServer() {
+              })
+
+              await session.closed
+              closed = true
+            }
+          },
+
+          // cleanly close remote sessions
+          async () => {
+            for await (const session of getReaderStream(
+              server.sessionStream('/session_close')
+            )) {
+              session.close()
+            }
+          },
+
+          // cleanly close remote sessions
+          async () => {
+            for await (const session of getReaderStream(
+              server.sessionStream('/session_close_with_reason')
+            )) {
+              session.close({
+                closeCode: 7,
+                reason: 'this is the reason'
+              })
+            }
+          },
+
+          // send data over unidirectional stream, initiated by remote
+          async () => {
+            for await (const session of getReaderStream(
+              server.sessionStream('/unidirectional_client_send')
+            )) {
+              const stream = await getReaderValue(
+                session.incomingUnidirectionalStreams
+              )
+              const received = await readStream(stream, KNOWN_BYTES.length)
+
+              // if we did not get the data we sent, close the session with a reason
+              if (!ui8.equals(ui8.concat(KNOWN_BYTES), ui8.concat(received))) {
+                session.close({
+                  closeCode: 500,
+                  reason: 'data did not match'
+                })
+              } else {
+                session.close()
+              }
+            }
+          },
+
+          // send data over unidirectional stream, initiated by local
+          async () => {
+            for await (const session of getReaderStream(
+              server.sessionStream('/unidirectional_server_send')
+            )) {
+              const stream = await session.createUnidirectionalStream()
+
+              await writeStream(stream, KNOWN_BYTES)
+            }
+          },
+
+          // delays reading from stream after client writes
+          async () => {
+            for await (const session of getReaderStream(
+              server.sessionStream('/unidirectional_server_delay_before_read')
+            )) {
+              const stream = await getReaderValue(
+                session.incomingUnidirectionalStreams
+              )
+
+              // wait before we read from the stream, should trigger backpressure
+              // on the client
+              await new Promise((resolve) => setTimeout(resolve, 1000))
+
+              const received = await readStream(stream, KNOWN_BYTES.length)
+
+              // if we did not get expected data, close the session with a reason
+              if (!ui8.equals(ui8.concat(KNOWN_BYTES), ui8.concat(received))) {
+                session.close({
+                  closeCode: 500,
+                  reason: 'data did not match'
+                })
+              } else {
+                session.close()
```

Comment:
Adding a time out before this line reduces the likelyhood of failure

Reply:
> // time out is needed, since it can be received before the read is complete

I'm not sure what this means?

The second argument to [readStream](https://github.com/fails-components/webtransport/pull/130/commits/2541994ab96fd480f940ff6f492482b78520d186#diff-a3f5b1d31bbe70635388f5f4f489039180ad6b6427a2179bd876ba9188b035afR208) is how many bytes it should wait to read so if we've hit his branch, it means we read the correct number of bytes but they had the wrong content, so waiting an extra second shouldn't change anything.

Reply:
Wait this is wrong. KNOWN_BYTES length is actually the number of chunks in KNOWN_BYTES .
As WebTransport does not gurantee, that the chunk are not merge or split, this cause the test occasionally to fail. So actually the number of bytes inside the chunks of KNOWN_BYTES added should be passed.